kernel/objects/capability-spinning/objects/CapEntry.hh

```diff
@@ -132,15 +132,24 @@ namespace mythos {
       uintlink_t val() const { return _val; }
       CapEntry* ptr() const { return _toPtr(); }
       bool has(uintlink_t flags) const { return (_val & flags) == flags; }
-      Link with(uintlink_t flags) const { return Link(_val & flags); }
```

Comment:
Yes, this is totally wrong. But the idea seemed to be, to avoid going through offset2kernel() and kernel2offset() for the pointer part. That got lost in your fix